src/main/java/org/folio/mr/service/impl/MediatedRequestActionsServiceImpl.java

```diff
@@ -264,6 +285,42 @@ private void extendMediatedRequest(MediatedRequestContext context) {
       .copyNumber(item.getCopyNumber());
   }
 
+  private void extendMediatedRequestWithInventoryItemDetails(MediatedRequestEntity mediatedRequestEntity) {
+    log.info("extendMediatedRequestWithInventoryItemDetails:: enhancing mediated request " +
+      "with additional item details");
+
+    searchService.searchInstance(mediatedRequestEntity.getInstanceId().toString())
+      .map(SearchInstance::getItems)
+      .stream()
+      .flatMap(List::stream)
+      .filter(searchItem -> searchItem.getId().equals(mediatedRequestEntity.getItemId().toString()))
+      .findFirst()
+      .ifPresent(searchItem -> {
+        String tenantId = searchItem.getTenantId();
+        executionService.executeAsyncSystemUserScoped(tenantId, () -> {
+          Item item = inventoryService.fetchItem(mediatedRequestEntity.getItemId().toString());
+          if (item == null) {
+            throw ExceptionFactory.notFound(format("Item %s not found",
+              mediatedRequestEntity.getItemId()));
+          } else {
+            log.info("extendMediatedRequestWithInventoryItemDetails:: item found, " +
+              "updating mediated request");
+            mediatedRequestEntity.setShelvingOrder(item.getEffectiveShelvingOrder());
+            ItemEffectiveCallNumberComponents components = item.getEffectiveCallNumberComponents();
+            if (components != null) {
+              mediatedRequestEntity.setCallNumber(components.getCallNumber());
+              mediatedRequestEntity.setCallNumberPrefix(components.getPrefix());
+              mediatedRequestEntity.setCallNumberSuffix(components.getSuffix());
+            }
+            String holdingsRecordId = item.getHoldingsRecordId();
+            if (holdingsRecordId != null) {
+              mediatedRequestEntity.setHoldingsRecordId(UUID.fromString(holdingsRecordId));
+            }
+          }
```

Comment:
Can we extract this block into a method?

Reply:
extracted